P11.py

```diff
@@ -0,0 +1,70 @@
+import threading
+
+class RWLock:
+    def __init__(self):
+        self.rw = 0
+        self.write_requests = 0
+        self.monitor = threading.Lock()
+        self.readers = threading.Condition(self.monitor)
+        self.writers = threading.Condition(self.monitor)
+
+    def read_lock (self):
+        self.monitor.acquire()

```

Comment:
acquire should only happen if the writer thread is present. Otherwise, we have no benefits